crates/context/config/src/client/env/config/types/starknet.rs

```diff
@@ -0,0 +1,545 @@
+use std::collections::BTreeMap;
+
+use hex;
+use starknet::core::codec::{Decode, Encode, Error, FeltWriter};
+use starknet::core::types::Felt;
+
+use crate::repr::{Repr, ReprBytes, ReprTransmute};
+use crate::types::SignerId;
+
+// Base type for all Starknet Felt pairs
+#[derive(Debug, Clone, Copy, Encode, Decode)]
+pub struct FeltPair {
+    pub high: Felt,
+    pub low: Felt,
+}
+
+// Newtype pattern following types.rs style
+#[derive(Debug, Clone, Copy, Encode, Decode)]
+pub struct ContextId(FeltPair);
+
+#[derive(Debug, Clone, Copy, Encode, Decode)]
+pub struct ContextIdentity(FeltPair);
+
+#[derive(Debug, Clone, Copy, Encode, Decode)]
+pub struct ApplicationId(FeltPair);
+
+#[derive(Debug, Clone, Copy, Encode, Decode)]
+pub struct ApplicationBlob(FeltPair);
+
+// Single conversion trait
+pub trait IntoFeltPair {
+    fn into_felt_pair(self) -> (Felt, Felt);
```

Comment:
for every `crate::types::{T}` here, you can impl `From<{T}>` for `FeltPair`, should come down to less code overall

the for the types here, it'll be for example

```rust
impl From<crate::types::ContextId> for ContextId {
    fn from(value: crate::types::ContextId) -> Self {
        Self(value.into())
    }
}
```